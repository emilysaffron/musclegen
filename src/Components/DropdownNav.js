import styled from "@emotion/styled";
import Menu from "./Menu";
import { useState } from "react";
const StyledModal = styled.div`
  display: flex;
  align-items: baseline;
  flex-direction: column;
  text-align: left;
  z-index: 100;
`;

const OpenMenu = styled.div`
  /* Is this font specific to the 'OPTIONS' text? If so fine, but if not you could apply this font-family value at a higher level, i.e.
  html {
    span {
      font-family: Palatino, URW Palladio L, serif;
    }
  }
   */
  font-family: Palatino, URW Palladio L, serif;
  font-size: 20px;
  color: black;
  cursor: pointer;

  align-self: baseline;
  align-items: center;
  flex-direction: column;
`;

const DropDownNav = () => {
  const [navState, toggleNavState] = useState(false);

  return (
    <div
      style={{ position: "fixed" }}
      onMouseEnter={() => toggleNavState(!navState)}
      onMouseLeave={() => toggleNavState(!navState)}
    >
      <OpenMenu onMouseEnter={() => toggleNavState(!navState)}>
        OPTIONS
      </OpenMenu>
      {navState ? (
        <StyledModal>
          <Menu
            labels={["Home", "New Workout", "Past Workouts", "Links"]}
            dropdown={true}
          />
        </StyledModal>
      ) : null}
    </div>
  );
};

export default DropDownNav;
