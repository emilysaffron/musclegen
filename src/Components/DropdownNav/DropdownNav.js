import styled from "@emotion/styled";
import Menu from "../Menu/Menu";
import { useState } from "react";
const StyledModal = styled.div`
  display: flex;
  align-items: baseline;
  flex-direction: column;
  text-align: left;
  z-index: 100;
`;

const OpenMenu = styled.div`
  font-size: 20px;
  cursor: pointer;
  align-self: baseline;
  align-items: center;
  flex-direction: column;
`;

const DropDownNav = () => {
  const [navState, toggleNavState] = useState(false);

  return (
    <div
      style={{ position: "fixed" }}
      onMouseEnter={() => toggleNavState(!navState)}
      onMouseLeave={() => toggleNavState(!navState)}
    >
      <OpenMenu onMouseEnter={() => toggleNavState(!navState)}>
        OPTIONS
      </OpenMenu>
      {navState ? (
        <StyledModal>
          <Menu
            options={["Home", "New Workout", "Past Workouts", "Videos"]}
            dropdown={true}
          />
        </StyledModal>
      ) : null}
    </div>
  );
};

export default DropDownNav;
