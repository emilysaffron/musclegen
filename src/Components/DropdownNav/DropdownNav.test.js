import renderer from "react-test-renderer";
import DropdownNav from "./DropdownNav";

it("DropdownNav Snapshot", () => {
  const tree = renderer.create(<DropdownNav />).toJSON();
  expect(tree).toMatchSnapshot();
});
