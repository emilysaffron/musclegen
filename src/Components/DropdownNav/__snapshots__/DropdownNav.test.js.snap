// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownNav Snapshot 1`] = `
<div
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "position": "fixed",
    }
  }
>
  <div
    className="css-pvpjm2"
    onMouseEnter={[Function]}
  >
    OPTIONS
  </div>
</div>
`;
